src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -350,17 +350,75 @@ private static MethodDefinition createTransportChannelProviderBeanMethod(
             .addDescription(nameStringAssignmentExpr)
             .build();
 
-    return MethodDefinition.builder()
-        .setHeaderCommentStatements(
-            SpringAutoconfigCommentComposer.createTransportChannelProviderComment())
-        .setName(methodName)
-        .setScope(ScopeNode.PUBLIC)
-        .setReturnType(STATIC_TYPES.get("TransportChannelProvider"))
-        .setAnnotations(
-            Arrays.asList(
-                AnnotationNode.withType(STATIC_TYPES.get("Bean")), conditionalOnMissingBean))
-        .setReturnExpr(returnExpr)
-        .build();
+    MethodDefinition.Builder beanMethodBuilder =
+        MethodDefinition.builder()
+            .setHeaderCommentStatements(
+                SpringAutoconfigCommentComposer.createTransportChannelProviderComment())
+            .setName(methodName)
+            .setScope(ScopeNode.PUBLIC)
+            .setReturnType(STATIC_TYPES.get("TransportChannelProvider"))
+            .setAnnotations(
+                Arrays.asList(
+                    AnnotationNode.withType(STATIC_TYPES.get("Bean")), conditionalOnMissingBean));
+
+    // LanguageServiceSettings.defaultTransportChannelProvider()
+    MethodInvocationExpr defaultTransportChannelProviderExpr =
+        MethodInvocationExpr.builder()
+            .setMethodName("defaultTransportChannelProvider")
+            .setStaticReferenceType(types.get("ServiceSettings"))
+            .setReturnType(STATIC_TYPES.get("TransportChannelProvider"))
+            .build();
+
+    // LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build()
+    Expr defaultTransportProviderExprChain =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(types.get("ServiceSettings"))
+            .setMethodName("defaultHttpJsonTransportProviderBuilder")
+            .build();
+    MethodInvocationExpr defaultHttpJsonTransportProviderExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(defaultTransportProviderExprChain)
+            .setMethodName("build")
+            .setReturnType(STATIC_TYPES.get("InstantiatingHttpJsonChannelProvider"))
+            .build();
+
+    if (hasRestOption) {
```

Comment:
This check is fine for preview release, but note that we need to come up with a strategy for REST only client like `compute`. Do we expect customers to always set this option to true or we automatically set this to true for REST only client? @zhumin8 @emmileaf

Reply:
I believe we are currently only populating `useRest` property in case `Transport.GRPC_REST`. So for REST only client, we could skip populating this property and set default to `useRest = True` behaviour